impeller/renderer/backend/gles/capabilities_gles.cc

```diff
@@ -109,6 +109,10 @@ CapabilitiesGLES::CapabilitiesGLES(const ProcTableGLES& gl) {
     default_glyph_atlas_format_ = PixelFormat::kR8UNormInt;
   }
 
+  if (desc->GetGlVersion().major_version >= 3) {
```

Comment:
Completely off topic, but we should do a GL version check in ProcTableGLES and reset or not bother to resolve [FOR_EACH_IMPELLER_GLES3_PROC](https://github.com/flutter/engine/blob/0a2034e0061f29dfc765ff6006d0e5b8fe397542/impeller/renderer/backend/gles/proc_table_gles.cc#L124) methods when we less than ES 3. On older versions of Android, I remember seeing instances where we'd get a function pointer for ES3 functions even on ES2 contexts but those wouldn't work if called.

When I first wrote that bit, I didn't think we'd ever resolve ES3 procs. So its a blind spot.

Reply:
https://github.com/flutter/flutter/issues/158995